fairscale/experimental/wgit/compression.py

```diff
@@ -0,0 +1,190 @@
+    """
+    ifft_state_dict = OrderedDict()
+    for name, weight in fft_state_dict.items():
+        if name in comp_layers:
+            # IFFT on the sparse complex topk and save only the real values
+            ifft_state_dict[name] = torch.fft.ifft(fft_state_dict[name]).real  # type: ignore
+        else:
+            ifft_state_dict[name] = weight
+    return ifft_state_dict
+
+
+def delta(state_dict: OrderedDict, recon_state_dict: OrderedDict) -> OrderedDict:
+    """Calculate the difference between the weights reconstructed from the FFT compressions and
+    subsequent iFFT.
+
+    Args:
+        state_dict (OrderedDict)
+            A model state_dict with the layer weight tensors of the dense model
+        recon_state_dict (OrderedDict)
+            A model state_dict with the layer weight tensors reconstructed by IFFT after having
+            been transformed with FFT at first.
+    """
+    delta_sd = OrderedDict()
+    for (name, weight), (name, weight_p) in zip(state_dict.items(), recon_state_dict.items()):
+        delta_sd[name] = weight - weight_p
+    return delta_sd
+
+
+def recons_from_del_sparse(delta_sps_sd: OrderedDict, inverse_recon_sd: OrderedDict) -> OrderedDict:
+    """Reconstruct the layer weights from the Delta Sparse Tensors and the IFFT Reconstructed tensors.
+    This function takes both the state_dict of Delta Sparse Tensors (DST) and the IFFT reconstructed
+    tensors as inputs. It returns a reconstructed state_dict from the inputs state_dicts.
+
+    Args:
+        delta_sps_sd (OrderedDict)
+            A model state_dict with the layer weight tensors of the dense model
+        recon_state_dict (OrderedDict)
+            A model state_dict with the layer weight tensors reconstructed by IFFT after having
+            been transformed with FFT at first.
+    """
+    recon_sps_sd = OrderedDict()
+    for (name, weight_s), (name, weight_p) in zip(delta_sps_sd.items(), inverse_recon_sd.items()):
+        recon_sps_sd[name] = weight_s + weight_p
+    return recon_sps_sd
+
+
+def linear_mask(w_tensor: Tensor, sparsity: float, dim: int = 1) -> Tensor:
+    """Generates a sparse mask for the input tensor based on the provided
+    sparsity level. The top-k connections of the linear layered are preserved
+    and the rest are pruned (zero-ed out).
+    a. dim = 0 results in a mask that selects the top-k neurons in the layer.
+    b. dim = 1 results in masks that selects top-k connections for each neuron.
+
+    Args:
+        w_tensor (Tensor)
+            A torch tensor representing a linear layer in a neural network.
+        sparsity (float)
+            Target sparsity of each tensors along the dimension dim.
+        dim (int)
+            The target dimension for which
+
+    """
+    thresholds_l = torch.quantile(w_tensor, q=sparsity, dim=1)  # type: ignore
+    linear_mask = w_tensor > thresholds_l.unsqueeze(1)
+    return linear_mask
+
+
+def compressible_layers(model: nn.Module) -> List[str]:
```

Comment:
@min-xu-ai this function currently seems to be quite ad-hoc. May be we can delegate the task of finding the list of layers to be compressed to the callers of other functions, and take an argument with the list?